pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -38,18 +39,15 @@ var (
 	Int64  = common.Int64
 )
 
-func (service *SecurityPolicyService) buildSecurityPolicyName(obj *v1alpha1.SecurityPolicy, createdFor string) string {
+func (service *SecurityPolicyService) buildSecurityPolicyName(obj *v1alpha1.SecurityPolicy) string {
 	if IsVPCEnabled(service) {
 		// For VPC scenario, we use obj.Name as the NSX resource display name for both SecurityPolicy and NetworkPolicy.
 		return util.GenerateTruncName(common.MaxNameLength, obj.Name, "", "", "", "")
 	}
-	prefix := common.SecurityPolicyPrefix
-	if createdFor != common.ResourceTypeSecurityPolicy {
-		prefix = common.NetworkPolicyPrefix
-	}
-	// For T1 scenario, we use ns-name as the key resource name for SecurityPolicy, it is to be consistent with the
+	// For T1 scenario, we use ns_name as the key resource name for SecurityPolicy, it is to be consistent with the
 	// previous solutions.
-	return util.GenerateTruncName(common.MaxNameLength, fmt.Sprintf("%s-%s", obj.Namespace, obj.Name), prefix, "", "", "")
+	return util.GenerateTruncName(common.MaxNameLength, util.ConnectStrings(common.ConnectorUnderline, obj.Namespace, obj.Name),
```

Comment:
I'd like to unify the name T1 and VPC, which will save our code maintenance work, also good for users. 
This is no rick for upgrade case if only chaing display name. we will not change IDs format for T1, so, it will not
change created NSX resources in T1 just update time. 

I will unify the name in the next patch when removing rule idx, so this patch will revert this change at moment.